napari/components/viewer_model.py

```diff
@@ -719,6 +719,82 @@ def add_shapes(
         self.add_layer(layer)
         return layer
 
+    def add_surface(
+        self,
+        data,
```

Comment:
I can't remember the outcome of the discussion about `data` vs arguments named after what they are (in this case, `vertices`)? I find this API confusing, because `data` doesn't actually capture the data we need to plot. If anything, `data` in this case should be the `vertices, faces, values` tuple (optionally, NamedTuple). Also, perhaps we can make values optional, and render a monochrome surface if desired?

Another (possibly headed straight to the bin) idea is to rename `faces` with `triangle_idxs` or something similarly descriptive. For non computer graphics people such as myself, this is more descriptive of what the value should be.

Similarly, `values` could be renamed to `face_values` or `triangle_values`.

Reply:
We can make data be a tuple - I'm not familiar with NamedTuples, so maybe not require that.

We could have values be a string for a color say `white` or `red` if someone wants monochrome, but then I think the api becomes more confusing and I'm not sure it's warranted. Similarly right now the `values` are actually `vertex_values` - there is one per vertex, we could instead also support `face_values` - one per face (I think) - or `vertex_colors` or `face_colors` which are `Nx4` arrays with actual colors instead of values to pass through a colormap. I feel like having `values` is probably sufficient, but I could rename as `vertex_values`.

Similarly I prefers `triangles` to `faces` and could do `triangle_idxs` if it was more clear.

We might need to talk this over, but I'm open to the suggestions

Reply:
NamedTuples are great! =)

```python
In [2]: from collections import namedtuple                                             
In [3]: namedtuple                                                                     
Out[3]: <function collections.namedtuple(typename, field_names, *, verbose=False, rename=False, module=None)>
In [5]: SurfaceMesh = namedtuple('SurfaceMesh', ['vertices', 'faces', 'values'])       
In [6]: data = SurfaceMesh(vertices=np.array([[0, 1], [1, 0], [0, 0]]), 
   ...:                    faces=np.array([[0, 1, 2]]), 
   ...:                    values=np.array([8.9]))                                     

In [7]: data[0]                                                                        
Out[7]: 
array([[0, 1],
       [1, 0],
       [0, 0]])

In [8]: data[1]                                                                        
Out[8]: array([[0, 1, 2]])

In [9]: data[2]                                                                        
Out[9]: array([8.9])

In [10]: data.vertices                                                                 
Out[10]: 
array([[0, 1],
       [1, 0],
       [0, 0]])

In [11]: data.faces                                                                    
Out[11]: array([[0, 1, 2]])

In [12]: data.values                                                                   
Out[12]: array([8.9])
```

Reply:
vertex_values is super confusing! Surely it makes more sense to color a face (which has extent) than a vertex (which does not)?

Reply:
A named tuple looks great, but i worry that making it the only input would be confusing to newcommers who have never used one before. We could convert from a normal tuple to a named tuple internally, but I still have a concern - about the standardization of the api.

For example in the #511 to zarr PR it's much easier if we have named keyword args that are simple objects like arrays instead of tuples

Reply:
Ok I have switched this over to accepting a tuple where within the layer model, the data gets split up into the different attributes. @kne42 I think might lean towards the names keyword arguments.

Also I hear you that coloring the faces might be more intuitive, but at least this dataset / file format came with the vertex values.

i've also renamed `values` as `vertex_values`